### ISSUE TITLE
Shadow jitter problem in large scenes

### ISSUE THREAD

**Issue by wddl** [google/filament#7241]
The 3D model is placed in a large world where the shadows shake at a Specific camera height. And this phenomenon does not occur in a small scene. How should this problem be solved?Thanks

_Originally posted by @wddl in https://github.com/google/filament/discussions/7188_

**Comment by wddl** [comment]
I have tested version 1.43.1, but the problem still exists. It's strange that the shadow effect is fine when the camera is at a far distance, not in the middle, and in close range the effect is fine.  stable is useful, but it has a limited effect, and a slightly larger scene is not. I put the model on the scale of the earth. Recorded a video and sent it out. There was a serious problem at a specific camera height.
![image](https://github.com/google/filament/assets/26518625/c2b728f8-08d0-45b2-8c6b-4b37a8a2b075)
![image](https://github.com/google/filament/assets/26518625/75c6a8be-d888-414c-9e1c-fd16688f6eee)
![image](https://github.com/google/filament/assets/26518625/0b850d70-6176-4753-94d2-95a72a057d64)

**Comment by romainguy** [comment]
@pixelflinger was working exactly on this problem. Do you have a repro case though?

**Comment by pixelflinger** [comment]
For very large scenes with the light behind the camera, you have to disable lispsm and use cascades instead.

This is because lispsm optimizes the resolution closer to the camera. In your picture, the light is almost behind the camera and the shadows are far from the camera.

If you want to keep lispsm you have to adjust the near plane hint in Shadow options.

Lispsm works great when the light is above and the direction perpendicular to the view direction.

**Comment by wddl** [comment]
@romainguy @pixelflinger Thanks to reply
I will test the method you said  @pixelflinger , but there are some incomprehensions, the light is sun type and not point light source, it only has one direction and no position. My key code is as follows, the shadow under this code is incorrect, you can refer to it. Hope to fix the bug ,thanks.

// Load Light
	m_sunLight = utils::EntityManager::get().create();
	LightManager::ShadowOptions shadowOptions;
	LightManager::Builder(LightManager::Type::SUN)
		.color(Color::toLinear<ACCURATE>({ 0.98f, 0.92f, 0.89f }))
		.intensity(100000)
		.direction(normalize(float3( 0.023456405464645523, -0.99952104730682578, -0.020185961293538123 ))) // light direction
		.sunAngularRadius(1.9f)
		.castShadows(true)
		.sunHaloSize(10.0)
		.sunHaloFalloff(80.0)
		.shadowOptions(shadowOptions)
		.build(*m_pFEngine, m_sunLight);
	
	// model entity, Model matrix
	auto& tcm = pEngine->getTransformManager();
	auto ti = tcm.getInstance(pAsset->getRoot());	
	mat4 yMat;	
	yMat[0] = { 1.0 , 0.0 , 0.0 , 0.0 };
	yMat[1] = { 0.0 , 0.0 , 1.0 , 0.0 };
	yMat[2] = { 0.0 , -1.0 , 0.0 , 0.0 };
	yMat[3] = { 0.0 , 0.0 , 0.0 , 1.0 };	
	mat4 localMat;
	localMat[0] = { -0.89529299190596490, -0.44547778692552764, 0.0000000000000000, 0.0000000000000000 };
	localMat[1] = { 0.28597298768987378, -0.57473036651310361, 0.76674927852554309, 0.0000000000000000  };
	localMat[2] = { -0.34156977172430397, 0.68646525561287353, 0.64194668305129443, 0.0000000000000000 };
	localMat[3] = { -2181624.2500000000, 4384490.0000000000, 4072700.0000000000, 1.0000000000000000 };
	tcm.setTransform(ti,    localMat * yMat);
	
	//look at
	FTransformManager& transformManager = EngineImp::instance()->m_pFEngine->getTransformManager();
	double3 feye(-2181820.2299841726  4385344.2504294114  4072977.9097391055);
	double3 fcenter(-2181819.888050911    4385343.5628314875    4072977.2691996922);
	double3 fup(0.28532232175636818   -0.57348227651259764  0.76792529013796906);
	transformManager.setTransform(transformManager.getInstance(m_pFcamera->getEntity()) , mat4::lookAt(feye, fcenter, fup));

**Comment by wddl** [comment]
@pixelflinger This is the model data.
[modern_city_block.zip](https://github.com/google/filament/files/12837069/modern_city_block.zip)

**Comment by pixelflinger** [comment]
Yes, I didn't choose the best words. I meant the distance from the eye to the shadowed point. I'm this case there is no shadow close to the camera, all shadows are far, and lipsm works against you in that case.

**Comment by wddl** [comment]
> Yes, I didn't choose the best words. I meant the distance from the eye to the shadowed point. I'm this case there is no shadow close to the camera, all shadows are far, and lipsm works against you in that case.
thanks！
I see, but in my case, the effect is good when the camera is far away from the shadow, but the effect is worse when the camera is closer, which is a bit confusing. Shadow effects get better when the camera is very close to the shadow. Shadows work well when the camera is at long distances and very close distances, but shadows don't work well at middle distances.

**Comment by pixelflinger** [comment]
Yes, it's because lispsm is able revert to an Ortho projection when it figures out that it's not working well. But that logic doesn't work all the time.

Far away it reverts to Ortho, so it looks okay, up close or does what is supposed to, so it works. In the middle, it struggles.

Just disable lispsm, or adjust the znear hint.

**Comment by wddl** [comment]
> Yes, it's because lispsm is able revert to an Ortho projection when it figures out that it's not working well. But that logic doesn't work all the time.
> 
> Far away it reverts to Ortho, so it looks okay, up close or does what is supposed to, so it works. In the middle, it struggles.
> 
> Just disable lispsm, or adjust the znear hint.

thanks！I'll try the method you mentioned
